gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java

```diff
@@ -54,9 +54,10 @@ public String getName() {
 
   public abstract String getBranchTypeString(boolean capitalized);
 
+  @SuppressWarnings("regexp")
   @Override
-  public GitCoreCommit getPointedCommit() throws GitCoreException {
-    GitCoreCommit gitCoreCommit = pointedCommitRef.get();
+  public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
+    var gitCoreCommit = pointedCommitRef.get();
     if (gitCoreCommit == null) {
       GitCoreCommit value = new GitCoreCommit(derivePointedRevCommit());
```

Comment:
I can see other places where `derivePointedRevCommit` is invoked... like in `deriveCommitsUntil`; but I don't think it should be called anywhere but just in this very line. `getPointedCommit` should be used instead to avoid double derivation. Also, for the same reason, `derivePointedRevCommit` can probably be private now.

Reply:
issue #281 created